### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by hughyoh**
Initial commit includes:

+ Project boilerplate
+ Mock data
+ Unit tests

### CODE REVIEW

File: middleware/auth.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  }
+
+  // Save the authorized vehicles for the credentials provided
+  req.vehicles = accounts[credentials];
+  return next();
+};
+
+/**
+ * Sends the vehicles retrivied from `basicAuthMiddleware`.
+ *
```
Comment (sankethkatta) [inline middleware/auth.js]: Put a `TODO` to say we'll implement real paging in the future.

File: middleware/mock_vehicles.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  if (!_.includes(req.vehicles, id)) {
+    return errors.sendAuthError(res, 'Unauthorized access for given vehicle.');
+  }
+
+  if (!_.has(mockVehicles, [id, path])) {
+    return next();
+  }
+
+  return res.send(mockVehicles[id][path]);
```
Comment (sankethkatta) [inline middleware/mock_vehicles.js]: do `res.json` instead to auto-set the proper `Content-Type` headers

File: middleware/auth.js (hunk 1 above)
Comment (hughyoh) [inline middleware/auth.js]: Done.

File: middleware/mock_vehicles.js (hunk 2 above)
Comment (hughyoh) [inline middleware/mock_vehicles.js]: Done.

**Pull request merged**